src/tmgrad.f90

```diff
@@ -38,22 +38,12 @@ subroutine tmgrad(nat,iat,xyz,g,edum)
    etemp = get_namespace('energy.tmp')
    open(newunit=ien,file=efile)
    open(newunit=jen,file=etemp)
-   read(ien,'(a)',end=33)a1
-33 continue
+   read(ien,'(a)')a1
    write(jen,'(a)')a1
-20 read(ien,'(a)',end=44)a
-   if(index(a,'$').eq.0)then
-      write(jen,'(a)')a
-      call readl(a,xx,nn)
-      j=idint(xx(1))
-   else
-   44 write(jen,'(i6,3F18.11)')j+1,edum,edum,edum
-      a='$end'
-      write(jen,'(a)')a
-      goto 22
-   endif
-   goto 20
-22 continue
+   read(ien,'(a)')a
+   write(jen,'(i6,3F18.11)')j+1,edum,edum,edum
+   a='$end'
+   write(jen,'(a)')a
```

Comment:
This refactoring is incorrect, the energy file usually looks like:
```
$energy
 1  <real> <real> <real>
 2  <real> <real> <real>
...
$end
```

The routine reads the file complete file and inserts a new line when the `$end` group is encountered.